package_scanner.nim

```diff
@@ -1,135 +1,367 @@
+  vcsTypes = ["git", "hg"]   ## Valid known Version Control Systems.
+  tagsBlacklist = ["nimrod", "nim"] ## Tags that should be not allowed.
+  keysRequired = ["name", "url", "method", "tags", "description", "license"]
+  gitTempDir = "/tmp/gitTempDir"
+  gitCmd = "git clone --no-checkout --no-tags --bare --depth 1 $1 " & gitTempDir
+  nodeCmd = "node ./validate_json.js"
+  pythonCmd = "python3 ./validate_json.py"
+  rubyCmd = "ruby ./validate_json.rb"
+  defaultFilePermissions = {fpUserWrite, fpUserRead, fpGroupWrite, fpGroupRead, fpOthersRead}
+  packagesFilePath = currentSourcePath().parentDir / "packages.json"
+  packagesJsonStr = readFile(packagesFilePath)
+  hostsSkip = [
+    "https://bitbucket",
+    "https://gitlab.3dicc",
+    "https://mahlon@bitbucket",
+    "https://notabug",
+  ] ## Hostnames to skip checks, for reliability, wont like direct HTTP GET?.
+  licenses = [
+    "allegro 4 giftware",
+    "apache",
+    "apache2",
+    "apache2.0",
+    "apache 2.0",
+    "apache-2.0",
+    "apache version 2.0",
+    "apache license 2.0",
+    "bsd",
+    "bsd2",
+    "bsd-2",
+    "bsd3",
+    "bsd-3",
+    "bsd 2-clause",
+    "bsd-2-clause",
+    "bsd 3-clause",
+    "bsd-3-clause",
+    "cc0",
+    "gpl",
+    "gpl2",
+    "gpl-2.0",
+    "gpl3",
+    "gpl-3.0",
+    "gplv2",
+    "gplv3",
+    "lgpl",
+    "lgplv2",
+    "lgplv2.1",
+    "lgplv3",
+    "mit",
+    "ms-pl",
+    "mpl",
+    "ppl",
+    "wtfpl",
     "libpng",
     "zlib",
-    "ISC",
-    "Unlicense"
-  ]
-  VCS_TYPES = @["git", "hg"]
+    "isc",
+    "unlicense",
+    # Not concrete Licenses, but some rare cases observed on the JSON.
+    "0bsd",
+    "agplv3",
+    "agpl-3.0",
+    "apache license 2.0 or mit",
+    "fontconfig",
+    "gnu lesser general public license v2.1",
+    "public domain",
+    "lgplv3 or gplv2",
+    "openssl and ssleay",
+    "openldap",
+    "apache 2.0 or gplv2",
+    "mit or apache 2.0",
+    "lgpl with static linking exception",
+  ]  ## All valid known licences for Nimble packages, on lowercase.
+when checkProfanity:
+  const profanitiesStr = readFile(currentSourcePath().parentDir / "profanities.txt").normalize.splitLines
+
+let
+  packagesJson = parseJson(packagesJsonStr).getElems         ## ``string`` to ``JsonNode``
+  pckgsList = filterIt(packagesJson, not it.hasKey("alias")) ## Packages, without Aliases
+  aliasList = filterIt(packagesJson, it.hasKey("alias"))     ## Aliases, without Packages
+  client = newHttpClient(timeout = httpTimeout) ## HTTP Client with Timeout
+  report = newJUnitOutputFormatter(openFileStream("report.xml", fmWrite)) ## JUnit Report XML
+addOutputFormatter(defaultConsoleFormatter())
+addOutputFormatter(report)
+when checkProfanity:
+  let profanities = filterIt(profanitiesStr, it.len > 0 and not(it.startsWith"#"))
+
+proc handler() {.noconv.} =
+  quit("CTRL+C Pressed, package_scanner is shutting down, Bye.")
+setControlCHook(handler)
+
+using url, name: string
```

Comment:
I personally dislike this construct, we shouldn't be using it.

Reply:
Why tho :grey_question: :thinking: Is just what the manual says, is something broken with the feature or something :grey_question: 
It wont say experimental nor deprecated.
Not like super important, I will remove it, just curious.